authz/middleware.go

```diff
@@ -67,9 +68,16 @@ func (mw *AuthzMiddleware) MiddlewareFunc(h rest.HandlerFunc) rest.HandlerFunc {
 			return
 		}
 
+		// extract resource action
+		action, err := extractResourceAction(r)
```

Comment:
the authz middleware is useless and awkward right now, keeping it because I'm lazy. but ok, I'll polish it till it shines.